### PR TITLE
docs: perf debounce the search query

### PR DESCRIPTION
**Pull request by shanpriyan**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
Debounced the search results to reduce the number of API calls

## Before fix

https://user-images.githubusercontent.com/42532987/204058551-4cd22983-34b1-4628-9132-4b0ae81494c1.mp4

## After fix
https://user-images.githubusercontent.com/42532987/204058560-ec2a2b9c-6683-4f92-a1b8-0639984d7b59.mp4

### CODE REVIEW

File: docs/src/assets/js/search.js [hunk 1]
```diff
@@ -114,6 +114,25 @@ function maintainScrollVisibility(activeElement, scrollParent) {
...
+ */
+function debounce(callback, delay) {
+    let timer;
+    return (...args) => {
+        if (timer) clearTimeout(timer)
+        timer = setTimeout(() => callback(...args), delay)
+    }
+}
+
+const debouncedFetchSearchResults = debounce((query, onsuccess, onerror) => {
```
Comment (amareshsm) [inline docs/src/assets/js/search.js:131]:
```suggestion
const debouncedFetchSearchResults = debounce((query, onSuccess, onError) => {
```

File: docs/src/assets/js/search.js [hunk 2]
```diff
@@ -114,6 +114,25 @@ function maintainScrollVisibility(activeElement, scrollParent) {
...
+    fetchSearchResults(query)
+        .then(onsuccess)
```
Comment (amareshsm) [inline docs/src/assets/js/search.js:133]:
```suggestion
        .then(onSuccess)
```

File: docs/src/assets/js/search.js [hunk 3]
```diff
@@ -114,6 +114,25 @@ function maintainScrollVisibility(activeElement, scrollParent) {
...
+        .catch(onerror);
```
Comment (amareshsm) [inline docs/src/assets/js/search.js:134]:
```suggestion
        .catch(onError);
```

File: docs/src/assets/js/search.js [hunk 4]
```diff
@@ -114,6 +114,25 @@ function maintainScrollVisibility(activeElement, scrollParent) {
...
+}, 300);
```
Comment (amareshsm) [inline docs/src/assets/js/search.js:135]:
```suggestion
const debouncedFetchSearchResults = debounce((query, onSuccess, onError) => {
    fetchSearchResults(query)
        .then(onSuccess)
        .catch(onError);
}, 300);
```

File: docs/src/assets/js/search.js [hunk 5]
```diff
@@ -114,6 +114,25 @@ function maintainScrollVisibility(activeElement, scrollParent) {
 
 }
 
+/**
+ * Debounces the provided callback with a given delay.
+ * @param {Function} callback The callback that needs to be debounced.
+ * @param {Number} delay Time in ms that the timer shoud wait before the callback is executed.
```
Comment (amareshsm) [inline docs/src/assets/js/search.js:120]:
```suggestion
 * @param {Number} delay Time in ms that the timer should wait before the callback is executed.
```

File: docs/src/assets/js/search.js [hunk 6]
```diff
@@ -114,6 +114,25 @@ function maintainScrollVisibility(activeElement, scrollParent) {
...
+ * @returns {Function}
+ */
+function debounce(callback, delay) {
+    let timer;
+    return (...args) => {
+        if (timer) clearTimeout(timer)
+        timer = setTimeout(() => callback(...args), delay)
```
Comment (amareshsm) [inline docs/src/assets/js/search.js:127]:
```suggestion
        timer = setTimeout(() => callback.apply(this, args), delay)
```
Use the apply method to bind the `this` & arguments to the called function

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

**Review by amareshsm — APPROVED** [review]
LGTM

**Review by harish-sethuraman — APPROVED** [review]

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**